src/config_map.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""A helper class for managing the configMaps holding the kratos config."""
+
+import logging
+from typing import Dict
+
+from lightkube import ApiError, Client
+from lightkube.models.meta_v1 import ObjectMeta
+from lightkube.resources.core_v1 import ConfigMap
+from ops.charm import CharmBase
+
+logger = logging.getLogger(__name__)
+
+
+class ConfigMapHandler(CharmBase):
```

Comment:
Sorry, I am confused about why we inherited the `CharmBase` class.

I guess it would be better if we layer in the following way:

|     kubeclient    |    configmap manager    |  charm    |

- kubeclient will be responsible for interacting with the k8s cluster using the `client` in the lightkube library. It will provide other callers with stable APIs, like `delete_config_map`, `patch_statefulsets`, etc. This will isolate the potential API changes from the third-party, and also we can easily swap the k8s client library in the future.
- configmap manager will be using the kubeclient APIs to compose biz logic layer, e.g. `create_all`, `delete_all`. 
- we extract the config map as either a protocol or base class with `create`, `update`, `delete` methods, and create subclasses inheriting the config map, e.g. `KratosConfigMap`, `ProviderConfigMap` to override the methods
- We can register all the types of the config map in the configmap manager, and then loop through the registry to create/delete them all in the `create/delete_all` methods. This way we can easily control the registry without changing the configmap manager's methods.
- This will give us scalability if more types of config maps show up in the future without changing configmap manager.

I think we could improve this later in other PRs.

Reply:
The `CharmBase` was left over, I removed it.

I tried to implement this abstraction except for the adapter around the lightkube client. We can work on that on a later PR if that's OK with you.